File: clients/lua/puid.lua
-- PUID client (Lua). luasocket + dkjson. Auth via PUID_API_KEY.
local http = require("socket.http")
local ltn12 = require("ltn12")
local json = require("dkjson")
local BASE = "https://puid.dev/api"
local M = {}
local function get(path)
  local chunks = {}
  local _, code = http.request{ url = BASE .. path,
    headers = { ["X-API-Key"] = os.getenv("PUID_API_KEY") or "" },
    sink = ltn12.sink.table(chunks) }
  if code == 429 then error("Rate limited. One per second.") end
  return json.decode(table.concat(chunks))
end
function M.generate(n) n = n or 1; assert(n >= 1 and n <= 10, "n must be 1..10"); return get("/v1/ids?n=" .. n).ids end
function M.ordinal(puid) return get("/v1/ordinal/" .. puid).ordinal end
return M

File: clients/ruby/puid.rb
# PUID client (Ruby). Auth via PUID_API_KEY.
require "net/http"; require "json"; require "uri"

module Puid
  BASE = "https://puid.dev/api"
  def self.get(path)
    uri = URI("#{BASE}#{path}")
    req = Net::HTTP::Get.new(uri); req["X-API-Key"] = ENV["PUID_API_KEY"]
    res = Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == "https") { |h| h.request(req) }
    raise "Rate limited. One per second." if res.code == "429"
    JSON.parse(res.body)
  end
  def self.generate(n = 1)
    raise ArgumentError, "n must be 1..10" unless (1..10).include?(n)
    get("/v1/ids?n=#{n}")["ids"]
  end
  def self.ordinal(puid) = get("/v1/ordinal/#{URI.encode_www_form_component(puid)}")["ordinal"].to_i
end
